logstash-core/lib/logstash/webserver.rb

```diff
@@ -105,6 +106,14 @@ def self.required_setting(settings, setting_name, trigger)
       settings.get(setting_name) || fail(ArgumentError, "Setting `#{setting_name}` is required when `#{trigger}` is set to `#{settings.get(trigger)}`. Please provide it in your `logstash.yml`")
     end
 
+    def self.required_setting_with_changing_default(settings, name, trigger, future_value)
+      effective_value = required_setting(settings, name, trigger)
+      if !settings.set?(name)
+        deprecation_logger.deprecated("The default value of `#{name}` will change to `#{future_value}` in a future release of Logstash. Set it to `#{future_value}` to observe the future behavior early, or set it to `#{effective_value}` to lock in the current behavior.")
```

Comment:
@karenzone can I get your review on the wording for this deprecation log?

As called it is effectively:

~~~
[2022-05-24T15:52:39,996][WARN ][deprecation.logstash.webserver] The default value of `api.auth.basic.password_policy.mode` will change to `ERROR` in a future release of Logstash. Set it to `ERROR` to observe the future behavior early, or set it to `WARN` to lock in the current behavior.
~~~

This deprecation log will be emitted when:

1) the user has explicitly configured `api.auth.type: basic`; AND
2) the user has _not_ specified `api.auth.basic.password_policy.mode`

We wish to change the _default_ value for `api.auth.basic.password_policy.mode` from its current `WARN` to `ERROR` at some undetermined point in the future (both options are currently supported). By specifying either value explicitly, the user can lock in their desired behavior and will be no longer at risk of the default changing out of under them, and will therefore not receive any more deprecation noise.

Reply:
A nice future-proof message. Thanks for the ping.